library/alloc/src/rc.rs

```diff
@@ -1168,7 +1168,7 @@ impl<T: ?Sized> Rc<T> {
     /// assert!(!Rc::ptr_eq(&five, &other_five));
     /// ```
     pub fn ptr_eq(this: &Self, other: &Self) -> bool {
-        this.ptr.as_ptr() == other.ptr.as_ptr()
+        ptr::eq(this.ptr.as_ptr() as *const (), other.ptr.as_ptr() as *const ())
```

Comment:
You can use `==` here, you don't need to call `ptr::eq`. The cast is the important part.